lang/cs/Org.Apache.REEF.Network/Group/Operators/IGroupCommOperator.cs

```diff
@@ -39,5 +39,10 @@ public interface IGroupCommOperator<T>
         /// The operator version number.
         /// </summary>
         int Version { get; }
+
+        /// <summary>
+        /// Ensure all parent and children nodes in the topology are registered with teh Name Service.
+        /// </summary>
+        void WaitForRegistration();
```

Comment:
Let me try to explain this. Before calling this function each task has to register itself with NameClient so that others can get out of their wait in the above function. Hence order of operation has to be: RegisterWithNetworkService followed by WaitForRegistration. Hence this cannot be called in operator constructor because then we have to call register before operator instantiation which will lead to Race Conditions. However, I think they can be called on the end of GroupCommClient constructor.